src/Core/src/Platform/iOS/CALayerAutoresizeToSuperLayer.cs

```diff
@@ -0,0 +1,166 @@
+#if IOS || MACCATALYST
+using System;
+using System.Runtime.InteropServices;
+using CoreAnimation;
+using CoreGraphics;
+using Foundation;
+using ObjCRuntime;
+
+namespace Microsoft.Maui.Platform;
+
+internal static class CALayerAutoresizeToSuperLayer
+{
+	static IntPtr _originalSetBoundsSelector;
+	static readonly IntPtr _associatedObjectKey;
+	static bool _initialized;
+
+	static CALayerAutoresizeToSuperLayer()
+	{
+		// Create a unique key for associated objects
+		_associatedObjectKey = Marshal.AllocHGlobal(1);
+	}
+
+	internal static void EnsureInitialized()
+	{
+		if (_initialized)
+		{
+			return;
+		}
+
+		_initialized = true;
+
+		// Get CALayer class handle
+		var caLayerClass = Class.GetHandle(typeof(CALayer));
+			
+		// Get the original setBounds: method
+		var setBoundsSelector = Selector.GetHandle("setBounds:");
+		var originalSetBoundsMethod = class_getInstanceMethod(caLayerClass, setBoundsSelector);
+			
+		if (originalSetBoundsMethod == IntPtr.Zero)
+		{
+			return;
+		}
+
+		// Create our swizzled method implementation
+		var swizzledSetBoundsImp = GetSwizzledSetBoundsImplementation();
+			
+		// Add our custom method to CALayer class
+		var mauiSetBoundsSelector = Selector.GetHandle("maui_setBounds:");
+			
+		// Try to add the method first
+		if (class_addMethod(caLayerClass, mauiSetBoundsSelector, swizzledSetBoundsImp, "v@:{CGRect={CGPoint=dd}{CGSize=dd}}"))
+		{
+			// Get the newly added method
+			var swizzledMethod = class_getInstanceMethod(caLayerClass, mauiSetBoundsSelector);
+				
+			// Exchange implementations
+			method_exchangeImplementations(originalSetBoundsMethod, swizzledMethod);
+			
+			// After swizzling, the original implementation is now at maui_setBounds:
+			_originalSetBoundsSelector = Selector.GetHandle("maui_setBounds:");
+		}
+	}
+
+	private static IntPtr GetSwizzledSetBoundsImplementation()
+	{
+		return Marshal.GetFunctionPointerForDelegate(new SetBoundsDelegate(Maui_SetBounds));
+	}
+
+	[UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+	private delegate void SetBoundsDelegate(IntPtr self, IntPtr selector, CGRect bounds);
+
+	[MonoPInvokeCallback(typeof(SetBoundsDelegate))]
+	private static void Maui_SetBounds(IntPtr self, IntPtr selector, CGRect bounds)
+	{
+		// Get the CALayer instance
+		var layer = Runtime.GetNSObject<CALayer>(self);
+			
+		if (layer == null)
+		{
+			return;
+		}
+
+		// Only if size has changed
+		var oldBounds = layer.Bounds;
+		if (!oldBounds.Equals(bounds))
+		{
+			var sublayers = layer.Sublayers;
+			if (sublayers != null)
+			{
+				foreach (var sublayer in sublayers)
+				{
+					if (GetAutoresizeToSuperLayer(sublayer))
```

Comment:
Given this is pure .NET compared to #30861 instead of using associated objects, we could simply check whether `sublayer` is some kind of `IAutoSizableCALayerInternal { AutoResize: true }`.